android-components/components/feature/search/src/main/java/mozilla/components/feature/search/telemetry/SERPTelemetryJsonParser.kt

```diff
@@ -0,0 +1,102 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.search.telemetry
+
+import android.content.Context
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import mozilla.components.support.base.log.logger.Logger
+import mozilla.components.support.ktx.android.org.json.toList
+import mozilla.components.support.remotesettings.RemoteSettingsClient
+import mozilla.components.support.remotesettings.RemoteSettingsResult
+import mozilla.components.support.remotesettings.readOrFetch
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import java.io.File
+
+/**
+ * Parse SERP Telemetry json from remote config.
+ */
+
+class SERPTelemetryJsonParser(context: Context) {
+
+    private val rootStorageDirectory: File = context.filesDir
+    private val coroutineScope = CoroutineScope(Dispatchers.IO)
+    private val searchProviders: Deferred<List<SearchProviderModel>> =
+        coroutineScope.async {
+            remoteSettingsDownloader()
+        }
+
+    suspend fun getSearchProviders(): List<SearchProviderModel> {
+        return searchProviders.await()
+    }
+
+    /**
+     * Fetches remote data from storage. If unavailable, fetches from the remote server.
+     */
+    private suspend fun remoteSettingsDownloader(): List<SearchProviderModel> {
+        val jsonFields: MutableList<SearchProviderModel> = mutableListOf()
+
+        val remoteSettingsClient = RemoteSettingsClient(
```

Comment:
I'm not sure how all this works together, so I'll ask anyway - what happens when Firefox is running and the collection gets updated on remote settings? Is there something that will update with the new records, or do we wait for the next startup?